x/ecocredit/server/core/query_credit_types.go

```diff
@@ -0,0 +1,28 @@
+package core
+
+import (
+	"context"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// CreditTypes queries the list of allowed types that credit classes can have.
+func (k Keeper) CreditTypes(ctx context.Context, _ *v1beta1.QueryCreditTypesRequest) (*v1beta1.QueryCreditTypesResponse, error) {
+	creditTypes := make([]*v1beta1.CreditType, 0)
+	it, err := k.stateStore.CreditTypeStore().List(ctx, ecocreditv1beta1.CreditTypePrimaryKey{})
```

Comment:
shouldn't we close the iterator as well?

Reply:
yup, thank you- updated here https://github.com/regen-network/regen-ledger/pull/847/commits/53945ae084d647f74c40aaf21f2dab7843aed4d5